tests/phpunit/tests/theme/wpAddGlobalStylesForBlocks.php

```diff
@@ -139,6 +139,79 @@ public function test_blocks_inline_styles_get_rendered() {
 		);
 	}
 
+	/**
+	 * @ticket 57868
+	 */
+	public function test_third_party_blocks_inline_styles_for_elements_get_rendered_when_per_block() {
+		$this->set_up_third_party_block();
+		add_filter( 'should_load_separate_core_block_assets', '__return_true' );
+
+		wp_register_style( 'global-styles', false, array(), true, true );
+		wp_enqueue_style( 'global-styles' );
+		wp_add_global_styles_for_blocks();
+
+		$actual = get_echo( 'wp_print_styles' );
+
+		$this->assertStringContainsString(
+			'.wp-block-my-third-party-block cite{color: white;}',
+			$actual
+		);
+	}
+
+	/**
+	 * @ticket 57868
+	 */
+	public function test_third_party_blocks_inline_styles_for_elements_get_rendered() {
+		wp_register_style( 'global-styles', false, array(), true, true );
+		wp_enqueue_style( 'global-styles' );
+		wp_add_global_styles_for_blocks();
+
+		$actual = get_echo( 'wp_print_styles' );
+
+		$this->assertStringContainsString(
+			'.wp-block-my-third-party-block cite{color: white;}',
+			$actual
+		);
+	}
+
+	/**
+	 * @ticket 57868
+	 * @dataProvider data_wp_get_block_name_from_theme_json_path
+	 */
```

Comment:
```suggestion
	/**
	 * @ticket 57868
	 *
	 * @dataProvider data_wp_get_block_name_from_theme_json_path
	 *
	 * @param array  $path     An array of keys describing the path to a property in theme.json.
	 * @param string $expected The expected block name.
	 */
```
Test method parameters should be documented